#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 20a0f3aa1b347862e4000ecde116ea6c5eec98a9 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Wed, 27 Mar 2019 12:53:10 +0100
Subject: [PATCH] Fix crash when looking at messages without attachs

* src/mapihelp.cpp (mapi_get_first_attach_data): Do not crash
if message has no attachments.
---
 src/mapihelp.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/mapihelp.cpp b/src/mapihelp.cpp
index 1b13076..4a4d0e9 100644
--- a/src/mapihelp.cpp
+++ b/src/mapihelp.cpp
@@ -488,6 +488,12 @@ mapi_get_first_attach_data (LPMESSAGE message)
     }

   mapi_attach_item_t *table = mapi_create_attach_table (message, 0);
+  if (!table)
+    {
+      log_debug ("%s:%s: Message has no attachments",
+                 SRCNAME, __func__);
+      TRETURN ret;
+    }
   if (table->end_of_table)
     {
       log_debug ("%s:%s: Message has no attachments",
--
2.20.1
